docs/sources/developers/plugins/migration-guide.md

```diff
@@ -4,77 +4,451 @@ title = "Plugin migration guide"
+)
+
+func main() {
+  log.DefaultLogger.Debug("Running GRPC server")
+
+  if err := datasource.Manage("myorgid-datasource", NewSampleDatasource, datasource.ManageOpts{}); err != nil {
+				log.DefaultLogger.Error(err.Error())
+				os.Exit(1)
+		}
+}
+```
+
+#### 3. Update the plugin package
+
+Update your `plugin` package to use the new plugin sdk.
+
+```go
+// before
+package plugin
+
+import (
+  "context"
+
+  "github.com/grafana/grafana_plugin_model/go/datasource"
+  "github.com/hashicorp/go-hclog"
+)
+
+func NewSampleDatasource(pluginLogger hclog.Logger) (*SampleDatasource, error) {
+	return &SampleDatasource{
+		logger: pluginLogger,
+	}, nil
+}
+
+type SampleDatasource struct{
+  logger hclog.Logger
+}
+
+func (d *SampleDatasource) Query(ctx context.Context, tsdbReq *datasource.DatasourceRequest) (*datasource.DatasourceResponse, error) {
+  d.logger.Info("QueryData called", "request", req)
+  // logic for querying your datasource.
+}
+
+// after
+package plugin
+
+import (
+  "context"
+
+  "github.com/grafana/grafana-plugin-sdk-go/backend"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/log"
+  "github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+func NewSampleDatasource(_ backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	return &SampleDatasource{}, nil
+}
+
+type SampleDatasource struct{}
+
+
+func (d *SampleDatasource) Dispose() {
+	// Clean up datasource instance resources.
+}
+
+func (d *SampleDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+  log.DefaultLogger.Info("QueryData called", "request", req)
+  // logic for querying your datasource.
+}
+
+func (d *SampleDatasource) CheckHealth(_ context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+  log.DefaultLogger.Info("CheckHealth called", "request", req)
+  // The main use case for these health checks is the test button on the
+  // datasource configuration page which allows users to verify that
+  // a datasource is working as expected.
+}
+```
+
+### Unsigned backend plugins will not be loaded
+
+Before you can run a backend plugin on Grafana installations, you must sign the plugin. Grafana Labs strongly recommends you upload signed plugins since we cannot guarantee the integrity of unsigned plugins.
+
+For instructions on plugin signing, refer to the [Plugin signatures](https://grafana.com/docs/grafana/latest/developers/plugins/sign-a-plugin/#sign-a-plugin) topic.
+
+You can still run and develop an unsigned plugin by running your Grafana instance in [development mode](https://grafana.com/docs/grafana/latest/administration/configuration/#app_mode).
+
+### Time series data can now be structured in a wide or many format.
+
+In Grafana 8.x, you can structure time series data in either `wide` or `many` formats. If your plugin supports time series data, we recommend that you support both formats.
```

Comment:
Is this for visualisations, or data source plugins?

Reply:
panel plugins

Reply:
Resolving this one as this part got removed from the guide. ([More info and explanation](https://github.com/grafana/grafana/pull/38911#discussion_r708992021))